Better handling of compilation failure warnings

uliska opened the pull request:
Follow-up to #77 (but slightly to involved to push directly to master)

With this commit only *one* warning/error is produced,
depending on whether an actual score has been generated or not.
'debug' and 'showfailed' options should work in any combination.

The compilation checks are now concentrated in one function,
un-cluttering run_lilypond() again.

jperon commented on lyluatex.lua:
```diff
@@ -237,12 +265,19 @@ function Score:check_failed_compilation()
         else
-            err("\nScore failed to compile, please check LilyPond input.\n")
+            err([[
+        Score failed to compile the score.
```
Wouldn't there be a typo here ? `s/Score/LilyPond/' ?

uliska commented:
ok, I'll fix and merge when I'm back home.
-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.

uliska commented:
OK, I've added some more coherence to the reporting and merged the branch. (Unfortunately I deleted the remote branch before pushing the merge, so this PR is labeled as "closed", not "merged".

In addition I added two test file in a new `examples` directory.
I've always tested with lots of random files, and I think it is a good idea to have a directory of example files that (ideally) cover all options and situations. These will not only serve as documentation but also as a sort of regression tests.
